server/src/test/java/org/opensearch/action/admin/indices/forcemerge/ForceMergeRequestTests.java

```diff
@@ -100,18 +102,26 @@ public void testBwcSerialization() throws Exception {
                 out.setVersion(compatibleVersion);
                 sample.writeTo(out);
 
-                final ForceMergeRequest deserializedRequest;
                 try (StreamInput in = out.bytes().streamInput()) {
                     in.setVersion(Version.CURRENT);
```

Comment:
I don't think this is the right way to test bwc. We want to generate the out serialized bytes using a compatible version. Then deserialize those bytes with the streamInput version as the generated compatible version only (this is because in version is controlling how the bytes were serialized from remote node). So if compatible version is < CURRENT, it will test the input in older format and see if current version is able to de-ser it correctly or not. You can refer [this](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/test/java/org/opensearch/action/admin/indices/close/CloseIndexRequestTests.java#L64)

Reply:
Gotcha, I've changed the code, please help to take a look, thank you~